### PR TITLE
refactor: use one result channel for all checks

### PR DESCRIPTION
**Pull request by lvlcn-t**
## Motivation

Addresses this discussion: https://github.com/caas-team/sparrow/pull/102#discussion_r1484805340

## Changes

I've removed the result channels of each check. Instead I'm giving the result channel to the `Run` method of each check.

For additional information look at the commits.

## Tests done

Unit tests succeeded.

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

File: pkg/checks/dns/dns_test.go [hunk 1]
```diff
@@ -190,13 +192,13 @@ func TestDNS_Run(t *testing.T) {
 				}
 			}()
 
-			r := <-c.ResultChan()
-			if r.Err != tt.want.Err {
-				t.Errorf("DNS.Run() = %v, want %v", r.Err, tt.want.Err)
+			r := <-cResult
+			if r.Result.Err != tt.want.Err {
+				t.Errorf("DNS.Run() = %v, want %v", r.Result.Err, tt.want.Err)
 			}
```
Comment (puffitos) [inline pkg/checks/dns/dns_test.go:198]: We have no case where an error is present in the result struct. Is this an omission on our part or does the `Result` not a good match for the DNS check? The Result.err is always empty in the `Run` function of the DNS check...

File: pkg/sparrow/controller.go [hunk 2]
```diff
@@ -47,17 +48,20 @@ func NewChecksController(dbase db.DB, metrics Metrics) *ChecksController {
...
+		cResult: make(chan checks.ResultDTO, 8), //nolint:gomnd // Buffered channel to avoid blocking the checks
 		cErr:    make(chan error, 1),
 		done:    make(chan struct{}, 1),
 	}
 }
 
-// HandleErrors handles errors that occur while running checks.
-func (cc *ChecksController) HandleErrors(ctx context.Context) error {
+// Run runs the ChecksController with handling results and errors.
+func (cc *ChecksController) Run(ctx context.Context) error {
```
Comment (puffitos) [inline pkg/sparrow/controller.go:58]: Finally, we have a nice and simple Run method. Well done.

**Review by puffitos — COMMENTED** [review]
LGTM. There is this `Err` field in the Result struct, which doesn't appear to be used anywhere. We should probably take a look at that.

File: pkg/checks/dns/dns_test.go (hunk 1 above)
Comment (puffitos) [reply pkg/checks/dns/dns_test.go:198]: Wanna remove it now? I don't feel like we should open an issue for this.

Comment (lvlcn-t) [reply pkg/checks/dns/dns_test.go]: Sure, I've removed it from the struct and all tests. 😄

**Pull request merged**